File: clients/python/retab/types/automations/webhooks.py
from typing import Any, Optional

from pydantic import BaseModel, EmailStr

from retab.types.documents.extractions import RetabParsedChatCompletion

from ..mime import BaseMIMEData, MIMEData


class WebhookRequest(BaseModel):
    completion: RetabParsedChatCompletion
    user: Optional[EmailStr] = None
    file_payload: MIMEData
    metadata: Optional[dict[str, Any]] = None


class BaseWebhookRequest(BaseModel):
    completion: RetabParsedChatCompletion
    user: Optional[EmailStr] = None
    file_payload: BaseMIMEData
    metadata: Optional[dict[str, Any]] = None

File: clients/node/src/types/automations/webhooks.ts
export interface WebhookRequest {
  completion: any; // RetabParsedChatCompletion
  user?: string;
  file_payload: any; // MIMEData
  metadata?: Record<string, any>;
}

export interface BaseWebhookRequest {
  completion: any; // RetabParsedChatCompletion
  user?: string;
  file_payload: any; // BaseMIMEData
  metadata?: Record<string, any>;
}